spring-integration-core/src/main/java/org/springframework/integration/channel/MessageChannelReactiveUtils.java

```diff
@@ -60,37 +61,18 @@ else if (messageChannel instanceof PollableChannel) {
 	}
 
 	private static <T> Publisher<Message<T>> adaptSubscribableChannelToPublisher(SubscribableChannel inputChannel) {
-		return new SubscribableChannelPublisherAdapter<>(inputChannel);
+		EmitterProcessor<Message<T>> publisher = EmitterProcessor.create(1);
+		@SuppressWarnings("unchecked")
+		MessageHandler messageHandler = (message) -> publisher.onNext((Message<T>) message);
+		return publisher
+				.doOnSubscribe((sub) -> inputChannel.subscribe(messageHandler))
+				.doOnCancel(() -> inputChannel.unsubscribe(messageHandler));
```

Comment:
Consider using `Flux.defer`, so that you return a cold `Publisher` that can be subscribed multiple times